drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -5943,12 +5943,22 @@ Vector<uint8_t> RenderingDeviceVulkan::buffer_get_data(RID p_buffer) {
 
 	VkCommandBuffer command_buffer = frames[frame].setup_command_buffer;
 
+	// Size of buffer to retreive.
+	if (!p_size) {
+		p_size = buffer->size;
+	} else {
+		ERR_FAIL_COND_V_MSG((p_size % 4) != 0, Vector<uint8_t>(),
+				"Size must be a multiple of four.");
```

Comment:
Where does this requirement come from?

Reply:
Mmm, I don't know what I was thinking, I'll remove it.